src/app/state/rest/useApiRequest.ts

```diff
@@ -45,55 +32,36 @@ const useApiRequest = <SCHEMA>({ url, group }: Config) => {
     return Promise.reject(details)
   }, [setError, addErrorFlashMessage])
 
-  /**
-   * Execute API request
-   */
+  // Execute an API request
   const exec = useCallback(async (method: Method, payload?: {}) => {
-    setIsBusy(true)
-
     try {
-      // collect cancelTokens, we want to be able to cancel all requests on unMount.
-      const cancelToken = axios.CancelToken.source()
-      cancelTokens.current.push(cancelToken)
-
-      const headers = {
-        Authorization: `Bearer ${ authorizationToken }`
-      }
-
-      const promise: Promise<AxiosResponse<SCHEMA>> =
-        // Is this request pending already?
-        pending[url] === undefined
-          // ...no its not! Execute request
-          ? api.request<SCHEMA>({
-              headers,
-              method,
-              url: slashSandwich([process.env.REACT_APP_GATEWAY, url]),
-              data: payload,
-              cache: { key: () => group },
-              cancelToken: cancelToken.token
+      setIsBusy(true)
+
+      const promise: Promise<AxiosResponse<SCHEMA>> = pending[url]
+        ?  pending[url]
+        :  axios.request<SCHEMA>({
+            headers: { Authorization: `Bearer ${ authorizationToken }` },
+            method,
+            url,
+            data: payload
           })
-          // ...yes it is, return the already pending request...
-          : pending[url]
 
-      // Save promise in pending object.
       pending[url] = promise
-
-      // Wait for the promise to resolve:
       const response = await promise
-      // Delete pending promise
       delete pending[url]
 
-      // Update state:
-      setData(response.data)
-      setIsBusy(false)
+      if (isMounted.current) {
+        method !== "get"
+          ? clear()
```

Comment:
Clear everything on POST/PUT/PATCH/DELETE for the sake of simplicity.
We can however reintroduce groups if we feel the need for it.